lib/puppet/functions/simplib/ip/family_hash.rb

```diff
@@ -0,0 +1,82 @@
+# Process an array of IP addresses and return them split by IP family and
+# include metadata and/or processed versions.
+#
+Puppet::Functions.create_function(:'simplib::ip::family_hash') do
+  # @param ip_addresses
+  #   The addresses to convert
+  #
+  # @return [Hash]
+  #   Converted Hash with the following format (YAML representation):
+  #
+  #   ```
+  #   # IPv4 Addresses
+  #   ipv4:
+  #     <Passed Address>:
+  #       address: <normalized address>
+  #       netmask:
+  #         ddq: <dotted quad notation netmask>
+  #         cidr: <CIDR netmask>
+  #   # IPv6 Addresses
+  #   ipv6:
+  #     <Passed Address>:
+  #       address: <normalized address>
+  #       netmask:
+  #         # DDQ is not valid for IPv6
+  #         ddq: nil
```

Comment:
why bother to set ddq if it is not applicable?

Reply:
Makes the dig logic easier when delving in using puppet language